File: src/main/scala/ComicsRss/Schedule.java
package ComicsRss;

import java.time.DayOfWeek;
import java.time.LocalDate;
import java.util.function.Function;

public enum Schedule {

	SevenDays(d -> true), MondayThroughSaturday(d -> d.getDayOfWeek() != DayOfWeek.SUNDAY), Sunday(d -> d.getDayOfWeek() == DayOfWeek.SUNDAY);

	private final Function<LocalDate, Boolean> acceptor;

	Schedule(final Function<LocalDate, Boolean> acceptor) {
		this.acceptor = acceptor;
	}

	public boolean accept(LocalDate date) {
		return acceptor.apply(date);
	}

}

File: src/main/kotlin/com/electronicmuse/comicsrss/Schedule.kt
package com.electronicmuse.comicsrss

import java.time.DayOfWeek
import java.time.OffsetDateTime

enum class Schedule(vararg days: DayOfWeek) {
    SEVEN_DAYS(DayOfWeek.SUNDAY, DayOfWeek.MONDAY, DayOfWeek.TUESDAY, DayOfWeek.WEDNESDAY, DayOfWeek.THURSDAY, DayOfWeek.FRIDAY, DayOfWeek.SATURDAY),
    SUNDAY(DayOfWeek.SUNDAY);

    val daySet = days.toSet()

    fun acceptDate(date: OffsetDateTime): Boolean {
        return daySet.contains(date.dayOfWeek)
    }
}
